### PR TITLE
Feature/nex 319/add lis parameters

### PR DESCRIPTION
**Pull request by siwane**
The purpose of this PR is to isolate the logic of LTILAUNCH build to allow to override it

### CODE REVIEW

File: controller/ToolModule.php [hunk 1]
```diff
@@ -115,4 +117,19 @@ public function launch()
...
+     *
+     * @param common_http_Request $request
+     * @return LtiLaunchData
+     * @throws ResolverException
+     */
+    protected function buildLaunchData(common_http_Request $request)
+    {
+        return LtiLaunchData::fromRequest(
+            $request,
+            [LtiLaunchData::LIS_OUTCOME_SERVICE_URL => _url('manageResults', 'ResultController', 'taoLtiConsumer')]
```
Comment (julien-sebire) [inline controller/ToolModule.php]: Why do we add LIS_OUTCOME_SERVICE_URL parameter and not LIS_RESULT_SOURCEDID in this call?

File: controller/ToolModule.php [hunk 2]
```diff
@@ -115,4 +117,19 @@ public function launch()
...
+        );
+    }
 }
```
Comment (julien-sebire) [inline controller/ToolModule.php]: Missing newline at end of file.

File: test/integration/controller/ToolModuleTest.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    public function run()
+    {
+    }
+
+    public function buildLaunchData(common_http_Request $request)
+    {
+        return parent::buildLaunchData($request);
+    }
+
+}
```
Comment (julien-sebire) [inline test/integration/controller/ToolModuleTest.php]: Missing newline at end of file.

File: test/integration/model/LtiLaunchDataTest.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+        $this->assertEmpty($data['customParams']);
+
+        $this->assertArrayHasKey('variables', $data);
+        $this->assertArrayHasKey('thisisatest', $data['variables']);
+        $this->assertEquals('predefinedValue', $data['variables']['thisisatest']);
+    }
+
+
+}
```
Comment (julien-sebire) [inline test/integration/model/LtiLaunchDataTest.php]: Missing newline at end of file.

**Comment by boajer** [comment]
Thank you @julien-sebire , your changes are awesome

**Pull request merged**